src/Fieldtypes/Date.php

```diff
@@ -341,9 +342,9 @@ public function toQueryableValue($value)
     private function parseSaved($value)
     {
         try {
-            return Carbon::createFromFormat($this->saveFormat(), $value);
+            return Carbon::createFromFormat($this->saveFormat(), $value)->tz($this->timezone());
```

Comment:
Is this approach wanted?

PRO: It's generic for all types of dates
CON: It works differently than the lastModified time:
https://github.com/statamic/cms/blob/4.x/src/Data/TracksLastModified.php#L13

The difference:
The implementation in this PR depends on the timezone from the config at `app.timezone`. Base on the defined timezone, the correct timezone will be calculated and saved in `UTC`. That works as expected.
Edge case: A user has defined the wrong timezone in his config, than the timestamp will be wrong.

We could use `Carbon::createFromTimestamp` instead, as used in the TracksLastModified trait. 
```php
if ($this->saveFormat() === "U") { 
    return Carbon::createFromTimestamp($value);
}
```

This works correctly, even with a wrong timezone in the config settings. 
The downside: This would only handle the saveFormat "U" and would not be a generic solution.

I think this is more a 'political' decision. Personally I am happy with both solutions. Let me know if you want a change.